alica_tests/src/test/test_alica_scheduling.cpp

```diff
@@ -0,0 +1,37 @@
+#include "test_alica.h"
+#include "CounterClass.h"
+
+#include <alica/test/Util.h>
+#include <gtest/gtest.h>
+
+namespace alica
+{
+namespace
+{
+
+class AlicaSchedulingPlan : public AlicaTestFixture
+{
+protected:
+    const char* getRoleSetName() const override { return "Roleset"; }
+    const char* getMasterPlanName() const override { return "SchedulingTestMasterPlan"; }
+    bool stepEngine() const override { return false; }
+};
+
+TEST_F(AlicaSchedulingPlan, scheduling)
```

Comment:
This test is just executing the engine without testing anything. So even if the order of execution is wrong this test would succeed. You need to run the engine in step mode and in between each step you need to check the right number of counters for each plan and method.